libs/sdk-ui-dashboard/src/model/commandHandlers/drill/resolveDrillToCustomUrl.ts

```diff
@@ -0,0 +1,243 @@
+                    ctx,
+                ),
+            );
+
+            return acc;
+        }, []),
+    );
+
+    return elements;
+}
+
+const encodeParameterIfSet = (parameter: string | undefined): string | undefined =>
+    parameter === undefined ? parameter : encodeURIComponent(parameter);
+
+function matchAll(regex: RegExp, text: string): RegExpExecArray[] {
+    const matches = [];
+    let match = null;
+    while ((match = regex.exec(text)) !== null) {
+        matches.push(match);
+    }
+    return matches;
+}
+
+export const getAttributeIdentifiersPlaceholdersFromUrl = (url: string): IDrillToUrlPlaceholder[] =>
+    matchAll(/{attribute_title\((.*?)\)}/g, url).map((match) => ({
+        placeholder: match[0],
+        identifier: match[1],
+        toBeEncoded: match.index !== 0,
+    }));
+
+export async function getAttributeDisplayForms(
+    projectId: string,
+    objRefs: ObjRef[],
+    ctx: DashboardContext,
+): Promise<IAttributeDisplayFormMetadataObject[]> {
+    return await ctx.backend.workspace(projectId).attributes().getAttributeDisplayForms(objRefs);
+}
+
+export function* getAttributeIdentifiersReplacements(
+    url: string,
+    drillIntersectionElements: IDrillEventIntersectionElement[],
+    ctx: DashboardContext,
+) {
+    const attributeIdentifiersPlaceholders = getAttributeIdentifiersPlaceholdersFromUrl(url);
+
+    if (attributeIdentifiersPlaceholders.length === 0) {
+        return [];
+    }
+
+    const displayForms: PromiseFnReturnType<typeof getAttributeDisplayForms> = yield call(
+        getAttributeDisplayForms,
+        ctx.workspace,
+        attributeIdentifiersPlaceholders.map((placeholder) => idRef(placeholder.identifier)),
+        ctx,
+    );
+
+    const elements: SagaReturnType<typeof loadAttributeElementsForDrillIntersection> = yield call(
+        loadAttributeElementsForDrillIntersection,
+        drillIntersectionElements,
+        displayForms,
+        ctx,
+    );
+
+    return attributeIdentifiersPlaceholders.map(({ placeholder: toBeReplaced, identifier, toBeEncoded }) => {
+        const elementTitle = elements.find(
+            (element: IDrillToUrlElement) => element.identifier === identifier,
+        )?.elementTitle;
+        const replacement = toBeEncoded ? encodeParameterIfSet(elementTitle) : elementTitle;
+
+        return {
+            toBeReplaced,
+            replacement,
+        };
+    });
+}
+
+const createIdentifierReplacement = (
+    toBeReplaced: DRILL_TO_URL_PLACEHOLDER,
+    replacement = "",
+): IDrillToUrlPlaceholderReplacement => ({ toBeReplaced, replacement, replaceGlobally: true });
+
+export function* getInsightIdentifiersReplacements(
+    customUrl: string,
+    widgetRef: ObjRef,
+    ctx: DashboardContext,
+) {
+    const { workspace } = ctx;
+    const dashboardId: ReturnType<typeof selectDashboardId> = yield select(selectDashboardId);
+    // const clientId = yield select(getWorkspaceClientId);
```

Comment:
This could probably be removed, right? as well as the other commented out code below.

Reply:
Or should we support this?